### ISSUE TITLE
Would like to have a way to override the configure script and avoid using sysctl

### ISSUE THREAD

**Issue by dmk42** [open-mpi/hwloc#300]
### What version of hwloc are you using?

1.11.9

### Which operating system and hardware are you running on?

SLES 12 on a Foxconn 64-bit ARM board
`uname -a` output:
Linux myhostname 4.4.73-5-default #1 SMP Tue Jul 4 15:33:39 UTC 2017 (b7ce4e4) aarch64 aarch64 aarch64 GNU/Linux

### Details of the problem

The configure script for hwloc makes an attempt to see if the `sysctl()` system call is available on the system.  However, it does not try very hard to make sure it actually works.  The most it does is try to link the following program to see if the linking succeeds.
```c
#include <stdio.h>
#include <sys/types.h>
#include <sys/sysctl.h>

int
main()
{
  return sysctl(NULL, 0, NULL, NULL, NULL, 0);
}
```

Meanwhile, SLES 12 on aarch64 provides a linkable version of `sysctl()` to allow legacy programs to be ported, but does not actually implement it.  Linking anything that calls `sysctl()` will result in the following warning message.
```
/tmp/cc2nRFTq.o: In function `main':
sysctl_example.c:(.text+0x20): warning: sysctl is not implemented and will always fail
```
The linker returns with exit status zero, so hwloc's configure script thinks `sysctl()` is available and proceeds to use it.

Because of this, any program that uses hwloc gets the above warning message from the linker.

It turns out that hwloc is doing the right thing by not trying harder to verify `sysctl()`, because sometimes it is impossible to run the test program at compile time, e.g. in the case of cross-compilation.

Therefore, what is needed is a way to override the configure script, and tell it that even though it looks like `sysctl()` is available, it should not be used.

### Additional information

This problem is being encountered by Chapel, which uses hwloc.  Every time a Chapel program is compiled for aarch64 on SLES 12, the linker warning appears, listing `hwloc_get_sysctl()` in src/topology.c line 123 as the function attempting to use it.  We would like for that not to happen.

**Comment by bgoglin** [comment]
Hello,
We got the same report two months ago. This is fixed in master since bab8388e0a6ccb47381159c345e9fdcb28cdab12 and in v1.11 since 58b867ac7de67d55922773b466bbfc3c15c43872 (we just don't look for sysctl on Linux anymore).
I'll release 1.11.10rc1 next week with the fix. Thanks

**Comment by dmk42** [comment]
Awesome, thanks!
